### PR TITLE
Add Tap to Pay icon on the transactions list page for Tap to Pay transactions

### LINKED ISSUE #5377: Add Tap to Pay logo to the transactions list view
In the Transactions list view (Payments > Transactions), make sure transactions made via Tap to Pay for iPhone have the correct channel and logo (still waiting for the design).

Transactions made with iPhone will have the following:

channel=`In-Person`
source=`ios`
source_identifier=`1234` (last 4 digits of the card used)

### PR DESCRIPTION
**Pull request by mordeth**
Fixes #5377

#### Changes proposed in this Pull Request
Adding a Tap to Pay icon on the transactions list page for Tap to Pay transactions.

#### Testing instructions
- Switch to the branch 3080-gh-Automattic/woocommerce-payments-server
- Create an order and pay it with Tap to Pay
- Navigate to Transactions list page
- Verify a Tap to Pay icon is displayed next to the `In Person` value in `Channel` column. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/data/transactions/hooks.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export interface Transaction {
 	// A field to identify the payment's source.
 	// Usually last 4 digits for card payments, bank name for bank transfers...
 	source_identifier: string;
+	source_device?: null | string;
```
Comment (anu-rock) [inline client/data/transactions/hooks.ts]: `source_device` is declared as a nullable property. Do we need to explicitly add `null` to its list of possible values?

**Review by anu-rock — APPROVED** [review]
✅ Code changes are neat. LGTM
✅ I was able to see the TTP icon.

image

File: client/data/transactions/hooks.ts (hunk 1 above)
Comment (elazzabi) [reply client/data/transactions/hooks.ts]: Agree, I think we can leave it out

File: client/transactions/list/index.tsx [hunk 2]
```diff
@@ -103,6 +104,18 @@ const getPaymentSourceDetails = ( txn: Transaction ) => {
 	}
 };
 
+const getSourceDeviceIcon = ( txn: Transaction ) => {
+	const tooltipDescription =
+		txn.source_device === 'ios'
+			? __( 'Tap to Pay on iPhone', 'woocommerce-payments' )
+			: __( 'Tap to Pay on Android', 'woocommerce-payments' );
```
Comment (elazzabi) [inline client/transactions/list/index.tsx]: I wonder if we should explicitly test for "android" and leave a default state for all non-ios and non-android devices.

```js
'ios' => __( 'Tap to Pay on iPhone', 'woocommerce-payments' )
'android' => __( 'Tap to Pay on Android', 'woocommerce-payments' )
default => __( 'Tap to Pay', 'woocommerce-payments' )
```

I think it's worth it given we can add server changes in the future without worrying of displaying incorrect information in the client.

File: client/transactions/list/index.tsx [hunk 3]
```diff
@@ -373,7 +386,12 @@ export const TransactionsList = (
 			},
 			channel: {
 				value: getChargeChannel( txn.channel ),
-				display: clickable( getChargeChannel( txn.channel ) ),
+				display: clickable(
+					<Fragment>
+						{ getChargeChannel( txn.channel ) }
+						{ txn.source_device ? getSourceDeviceIcon( txn ) : '' }
```
Comment (elazzabi) [inline client/transactions/list/index.tsx]:
```suggestion
						{ txn.source_device && getSourceDeviceIcon( txn ) }
```

**Review by elazzabi — CHANGES_REQUESTED** [review]
Thanks for working on this! I added a few minor comments 🙇 

To test this locally without having Tap to Pay setup, you can "force" the `source_device` using:

```js
// Include in client/transactions/list/index.tsx, line 372
txn.source_device = 'ios';
```

File: client/transactions/list/index.tsx (hunk 3 above)
Comment (anu-rock) [reply client/transactions/list/index.tsx]: I had the same suggestion in mind and then forgot all about it :p

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5744` or branch name `add/ttp-logo-transactions-page` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/ttp-logo-transactions-page) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 6d2746c800466e8e4056af470ad8b3c1a360594e
- Build time: 2023-03-23 06:31:23 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/transactions/list/index.tsx (hunk 3 above)
Comment (mordeth) [reply client/transactions/list/index.tsx]: Good catch, updated! :)

File: client/transactions/list/index.tsx (hunk 2 above)
Comment (mordeth) [reply client/transactions/list/index.tsx]: It makes sense, updated.

File: client/data/transactions/hooks.ts (hunk 1 above)
Comment (mordeth) [reply client/data/transactions/hooks.ts]: Updated!

**Review by elazzabi — APPROVED** [review]

**Pull request merged**